hot-reload-agent/src/main/kotlin/org/jetbrains/compose/reload/agent/recompiler.kt

```diff
@@ -35,34 +41,61 @@ private val recompileRequests = LinkedBlockingQueue<RecompileRequest>(
 )
 
 internal fun launchRecompiler() {
-    val gradleBuildRoot = gradleBuildRoot ?: run {
-        logger.error("Missing 'compose.build.root' property")
-        return
-    }
-
-    val gradleBuildProject = gradleBuildProject ?: run {
-        logger.error("Missing 'compose.build.project' property")
-        return
-    }
-
-    val gradleBuildTask = gradleBuildTask ?: run {
-        logger.error("Missing 'compose.build.compile.task' property")
-        return
-    }
 
     val port = ComposeHotReloadAgent.orchestration.port
     logger.debug("'Compose Recompiler': Using orchestration at '$port'")
 
+    val processBuilder = when (buildSystem ?: "gradle") {
+        "amper" -> {
```

Comment:
What do you think about modelling this as enum, so that the name of the enum can be easily used as param?